libcarina_test.go

```diff
@@ -0,0 +1,223 @@
+
+	carinaClient, err := createMockCarinaClient(mockIdentity.URL+"/v2.0/", mockCarina.URL)
+	if err != nil {
+		t.Error("wasn't able to create carinaClient pointed at mockCarina.URL with error:", err)
+		t.FailNow()
+	}
+	resp, err := carinaClient.ListClusterTypes()
+	if resp != nil {
+		t.Error("expected nil response, got", resp)
+	}
+	assertMicroversionUnsupportedHandled(t, err)
+}
+
+func TestMicroversionUnsupportedCreate(t *testing.T) {
+	mockCarina, mockIdentity := createMockCarina(microversionUnsupportedHandler)
+	defer mockCarina.Close()
+	defer mockIdentity.Close()
+
+	carinaClient, err := createMockCarinaClient(mockIdentity.URL+"/v2.0/", mockCarina.URL)
+	if err != nil {
+		t.Error("wasn't able to create carinaClient pointed at mockCarina.URL with error:", err)
+		t.FailNow()
+	}
+	clusterOpts := &CreateClusterOpts{
+		Name:          "test-cluster",
+		ClusterTypeID: 1,
+		Nodes:         2,
+	}
+	resp, err := carinaClient.Create(clusterOpts)
+	if resp != nil {
+		t.Error("expected nil response, got", resp)
+	}
+	assertMicroversionUnsupportedHandled(t, err)
+}
+
+func TestMicroversionUnsupportedDelete(t *testing.T) {
+	mockCarina, mockIdentity := createMockCarina(microversionUnsupportedHandler)
+	defer mockCarina.Close()
+	defer mockIdentity.Close()
+
+	carinaClient, err := createMockCarinaClient(mockIdentity.URL+"/v2.0/", mockCarina.URL)
+	if err != nil {
+		t.Error("wasn't able to create carinaClient pointed at mockCarina.URL with error:", err)
+		t.FailNow()
+	}
+	resp, err := carinaClient.Delete("9f18f7f9-aeb4-4c7c-91ef-e13ff94e352c")
+	if resp != nil {
+		t.Error("expected nil response, got", resp)
+	}
+	assertMicroversionUnsupportedHandled(t, err)
+}
+
+func TestMicroversionUnsupportedResize(t *testing.T) {
+	mockCarina, mockIdentity := createMockCarina(microversionUnsupportedHandler)
+	defer mockCarina.Close()
+	defer mockIdentity.Close()
+
+	carinaClient, err := createMockCarinaClient(mockIdentity.URL+"/v2.0/", mockCarina.URL)
+	if err != nil {
+		t.Error("wasn't able to create carinaClient pointed at mockCarina.URL with error:", err)
+		t.FailNow()
+	}
+	resp, err := carinaClient.Resize("9f18f7f9-aeb4-4c7c-91ef-e13ff94e352c", 3)
+	if resp != nil {
+		t.Error("expected nil response, got", resp)
+	}
+	assertMicroversionUnsupportedHandled(t, err)
+}
+
+func TestMicroversionUnsupportedGetCredentials(t *testing.T) {
+	mockCarina, mockIdentity := createMockCarina(microversionUnsupportedHandler)
+	defer mockCarina.Close()
+	defer mockIdentity.Close()
+
+	carinaClient, err := createMockCarinaClient(mockIdentity.URL+"/v2.0/", mockCarina.URL)
+	if err != nil {
+		t.Error("wasn't able to create carinaClient pointed at mockCarina.URL with error:", err)
+		t.FailNow()
+	}
+	resp, err := carinaClient.GetCredentials("9f18f7f9-aeb4-4c7c-91ef-e13ff94e352c")
+	if resp != nil {
+		t.Error("expected nil response, got", resp)
+	}
+	assertMicroversionUnsupportedHandled(t, err)
+}
+
+func TestMicroversionUnsupportedGetAPIMetadata(t *testing.T) {
```

Comment:
I would lean towards removing this as part of the story, since it's not used anymore.